src/transport/PeerMessageCounter.h

```diff
@@ -85,7 +86,65 @@ class PeerMessageCounter
         return CHIP_NO_ERROR;
     }
 
-    CHIP_ERROR Verify(uint32_t counter) const
+    /**
+     * @brief Implementation of spec 4.5.4.2
+     *
+     * For encrypted messages of Group Session Type, any arriving message with a counter in the range
+     * [(max_message_counter + 1) to (max_message_counter + 2^31 - 1)] (modulo 2^32) SHALL be considered
+     * new, and cause the max_message_counter value to be updated. Messages with counters from
+     * [(max_message_counter - 2^31) to (max_message_counter - MSG_COUNTER_WINDOW_SIZE - 1)] (modulo 2^
+     * 32) SHALL be considered duplicate. Message counters within the range of the bitmap SHALL be
+     * considered duplicate if the corresponding bit offset is set to true.
+     *
+     */
+    CHIP_ERROR VerifyGroupCounter(uint32_t counter)
+    {
+        uint32_t highLimit = mSynced.mMaxCounter + kGroupOffset + 1;
+        // 1. Check counter value if in valid range
+        if (highLimit < mSynced.mMaxCounter)
+        {
+            // Rollover possible
+            if ((counter > mSynced.mMaxCounter) || (counter <= highLimit))
+            {
+                return CHIP_NO_ERROR;
+            }
+        }
+        else
+        {
+            if ((counter > mSynced.mMaxCounter) && (counter <= highLimit))
+            {
+                return CHIP_NO_ERROR;
+            }
+        }
+        // 2.  Check counter if in invalid range
+        uint32_t invalidLowerLimit = mSynced.mMaxCounter - kGroupOffset;
+        highLimit                  = mSynced.mMaxCounter - CHIP_CONFIG_MESSAGE_COUNTER_WINDOW_SIZE - 1;
+        if (highLimit > invalidLowerLimit)
```

Comment:
Why do we need this check?  Why isn't just testing that `mSynced.mMaxCounter - counter < CHIP_CONFIG_MESSAGE_COUNTER_WINDOW_SIZE` (which we end up doing below) enough?

Reply:
Removed